yt_dlp/extractor/googledrive.py

```diff
@@ -237,14 +237,11 @@ def add_source_format(urlh):
                     urlh, url, video_id, note='Downloading confirmation page',
                     errnote='Unable to confirm download', fatal=False)
                 if confirmation_webpage:
-                    confirm = self._search_regex(
-                        r'confirm=([^&"\']+)', confirmation_webpage,
+                    confirmed_source_url = self._html_search_regex(
+                        r'action="(.+?)"', confirmation_webpage,
```

Comment:
Should I use `get_element_html_by_id` to obtain the specific element I want so that the regex doesn't get confused with the login page, like `get_element_html_by_id('download-form', confirmation_webpage)`?

Reply:
good idea IMO. you could also use `extract_attributes`, e.g.
```py
                    confirmed_source_url = extract_attributes(
                        get_element_html_by_id('download-form', confirmation_webpage) or '').get('action')
```